syft/generic/frameworks/hook/hook.py

```diff
@@ -567,3 +636,121 @@ def overloaded_attr(self, *args, **kwargs):
+           '__add__' of the 'str' type. 
+           That is why the 'String' argument to '__add__' should
+           be peeled down to 'str'
+        
+           Args:
+               args: A tuple or positional arguments of the method
+                     being hooked to the String class.
+
+           Return:
+               A list of adapted positional arguments.
+           
+        """
+
+        new_args = []
+
+        for arg in args:
+
+            # If 'arg' is an object of type String
+            # replace it by and 'str' object
+            if isinstance(arg, String):
+                new_args.append(arg.child)
+            else:
+                new_args.append(arg)
+
+        return new_args
+
+    @classmethod
+    def _wrap_str_return_value(cls, self, attr: str, value: object):
+
+        # The outputs of the following attributed won't
+        # be wrapped
+        ignored_attr = set(["__str__", "__repr__", "__format__"])
+
+        if isinstance(value, str) and attr not in ignored_attr:
+
+            return String(object=value, owner=self.owner)
+
+        return value
+
+    @classmethod
+    def _get_hooked_string_method(cls, attr):
+        """
+           Hook a `str` method to a corresponding method  of 
+          `String` with the same name.
+
+           Args:
+               attr (str): the method to hook
+           Return:
+               the hooked method
+
+        """
+
+        @wraps(attr)
+        def overloaded_attr(self, *args, **kwargs):
+
+            args = cls._string_input_args_adaptor(args)
+
+            # Call the method of the core builtin type
+            native_response = getattr(self.child, attr)(*args, **kwargs)
+
+            # Some return types should be wrapped using the String
+            # class. For instance, if 'foo' is an object of type
+            # 'String' which wraps 'str'. calling foo.upper()
+            # should also be of type 'String' not 'str'.
+            # However, the return value of foo.__str__ should
+            # be of type 'str'.
+            response = cls._wrap_str_return_value(self, attr, native_response)
+
+            return response
+
+        return overloaded_attr
+
+    @classmethod
+    def _get_hooked_string_pointer_method(cls, attr):
+        """
+           Hook a `String` method to a corresponding method  of 
+          `StringPointer` with the same name.
+
+           Args:
+               attr (str): the method to hook
+           Return:
+               the hooked method
+
+        """
+
+        @wraps(attr)
+        def overloaded_attr(self, *args, **kwargs):
+            """
+            Operate the hooking
+            """
+
+            owner = self.owner
+            location = self.location
+            id_at_location = self.id_at_location
+
+            # Create a 'command' variable  that is understood by
+            # the send_command() method of a worker.
+            command = (attr, id_at_location, args, kwargs)
```

Comment:
usually it is expected to put "self" and not "self.id_at_location"

Reply:
That is right, Done!